config/accounts.go

```diff
@@ -1,9 +1,61 @@
 package config
 
+// RequestType : Request type enumeration
+type RequestType string
+
+// The request types
+const (
+	RequestTypeAMP   RequestType = "AMP"
+	RequestTypeApp   RequestType = "app"
+	RequestTypeVideo RequestType = "video"
+	RequestTypeWeb   RequestType = "web"
+)
+
 // Account represents a publisher account configuration
 type Account struct {
-	ID            string      `mapstructure:"id" json:"id"`
-	Disabled      bool        `mapstructure:"disabled" json:"disabled"`
-	CacheTTL      DefaultTTLs `mapstructure:"cache_ttl" json:"cache_ttl"`
-	EventsEnabled bool        `mapstructure:"events_enabled" json:"events_enabled"`
+	ID            string      `json:"id"`
+	Disabled      bool        `json:"disabled"`
+	CacheTTL      DefaultTTLs `json:"cache_ttl"`
+	EventsEnabled bool        `json:"events_enabled"`
+	GDPR          AccountGDPR `json:"gdpr"`
+}
+
+// AccountGDPR represents account-specific GDPR configuration
+type AccountGDPR struct {
+	Enabled            *bool                  `json:"enabled,omitempty"`
+	IntegrationEnabled AccountGDPRIntegration `json:"integration_enabled"`
+}
+
+// EnabledForRequestType indicates whether GDPR is turned on at the account level for the specified request type
+// by using the request type setting if defined or the general GDPR setting if defined; otherwise it returns nil
+func (a *AccountGDPR) EnabledForRequestType(requestType RequestType) *bool {
+	var integrationEnabled *bool
```

Comment:
I feel this additional setting has potential for ambiguous situations. For example, [enabled=true, all others=false], or [enabled=false, all others=true]. With account defaults layered in, activation of the individual settings is even less clear.

What are your thoughts on removing this fallback altogether ? We do not lose any configurability as the behavior can be fully specified by the four settings in the same section, and overrides apply straight on each field.

Reply:
@hhhjort @SyntaxNode My take was that the committee already agreed on the configuration settings so it would be preferred to follow that, though I understand the two versions sometimes deviate from one another. I recognize that potential ambiguity @laurb9 pointed out. How do you guys feel about this?

Regarding account defaults, I figured the defaults for all of these is `nil`; you either need to explicitly set these on the account or fall back to whatever the host settings are.

Reply:
Agreed that those cases are ambiguous. The intent is to provide an easy on/off switch with the ability to fine tune it if needed per request type. I suppose this is geared toward expressiveness in config. If an account doesn't care about each integration type, it's easier to set one enabled flag instead of four. The fallback also better defines future compatibility if a new integration type is added in the future.

Since this is implementing an already approved spec, I recommend moving this conversation to https://github.com/prebid/prebid-server/issues/1323.

I'd prefer to keep the same settings as PBS-Java currently for simplicity's sake.